Automattic/wp-calypso
# My Plan: Display storage usage on Backup purchase cards

Resolves `1200412004370260-as-1200532973279692`.

#### Changes proposed in this Pull Request

* Add specific styling rules to the `MyPlanCard` component to accommodate inclusion of the backup storage display.
* Align sections of the `MyPlanCard` component to align themselves to the start of the block axis (i.e., the top) instead of centering, so that the secondary section still lines up with the product name.
* Move the `BackupStorageUsage` component from `calypso/my-sites/backup` to `calypso/components`, since it's now used in more places than the Backup page.
* Add logic to `BackupStorageUsage` to generate an appropriate upsell URL for both Calypso Blue and Calypso Green.

#### Known issues

* The storage usage display is currently built from mock data. Once an API endpoint is available to retrieve real, site-specific data, we'll open up another PR to start using it.
* Clicking the upsell button currently redirects to the generic Plans page. Eventually it will redirect to a Backup-specific tier selection page.

#### Testing instructions

1. Open Calypso Blue and select a Jetpack site with an active Backup subscription.
2. Select **Upgrades > Plans** from the navigation sidebar; then, when the page loads, select the **My Plan** tab.
3. Verify you see your selected site's Backup purchase in the purchases listing, and that no storage information is displayed.
4. Add `?flags=jetpack/only-realtime-products` to your browser URL and reload the page.
5. Verify that on the lower portion of the card display for your site's Backup purchase, you now see storage usage information.
6. Resize the browser window to confirm the screen layout works for both mobile and desktop devices (see "Reference capture").
7. Click the "Upgrade your storage" upsell button, verifying that you're redirected to the Plans page.

#### Reference capture

https://user-images.githubusercontent.com/670067/128225088-bd71c7a5-fb8d-4855-b6d4-a79c0a8bb113.mp4
